### PR TITLE
[#1158] add test coverage for QuickCreate widget

### PR DESCRIPTION
**Pull request by thumbsupep**
# Description

Add test coverage for QuickCreate widget

## More Details

wrote unit and integration tests for 100% coverage
Screen Shot 2019-10-05 at 5 16 41 PM

## Corresponding Issue

https://github.com/ifmeorg/ifme/issues/1158

---

Reviewing this pull request? Check out our [Code Review Practices](https://github.com/ifmeorg/ifme/wiki/Code-Review-Practices) guide if you haven't already!

### CODE REVIEW

File: client/app/widgets/QuickCreate/_tests_/QuickCreate.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+      wrapper.update();
+    }
+
+    it('opens modal and displays quick create form if label does not exist', () => {
+      changeInput('new_label');
+      expect(wrapper.find('Modal').prop('open')).toBeTruthy();
+      expect(wrapper.find('DynamicForm')).toHaveLength(1);
+    });
+
+    it('does nothing if label exists', () => {
```
Comment (julianguyen) [inline client/app/widgets/QuickCreate/_tests_/QuickCreate.spec.jsx]: I would rephrase this to, 'does not open the modal if the label already exists'.

File: client/app/widgets/QuickCreate/_tests_/QuickCreate.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+      changeInput('alpha_label');
+      expect(wrapper.find('Modal').prop('open')).toBeFalsy();
+      expect(wrapper.find('DynamicForm')).toHaveLength(0);
+    });
+  });
+
+  describe('when dynamic form is submit', () => {
```
Comment (julianguyen) [inline client/app/widgets/QuickCreate/_tests_/QuickCreate.spec.jsx]: Let's rephrase this to, 'when the form is submitted'.

File: client/app/widgets/QuickCreate/_tests_/QuickCreate.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+      submitForm();
+      input = wrapper.find('Input');
+      modal = wrapper.find('Modal');
+      expect(modal.prop('open')).toBeFalsy();
+      expect(input.prop('accordionOpen')).toBeTruthy();
+      expect(modal.key()).toBe(wrapper.state('modalKey'));
+      expect(input.key()).toBe(wrapper.state('tagKey'));
+    });
+
+    it('does not do anything if response data is not a success', () => {
```
Comment (julianguyen) [inline client/app/widgets/QuickCreate/_tests_/QuickCreate.spec.jsx]: Let's rephrase this to, 'does not submit the form when the response is invalid'.

File: client/app/widgets/QuickCreate/_tests_/QuickCreate.spec.jsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+    it('adds checkboxes from data', () => {
+      const spy = jest.spyOn(wrapper.instance(), 'addToCheckboxes');
+      expect(wrapper.find('Input').prop('checkboxes')).toHaveLength(4);
+      submitForm();
+      expect(wrapper.find('Input').prop('checkboxes')).toHaveLength(5);
+      expect(spy).toHaveBeenCalledTimes(1);
+    });
+
+    it('opens accordion and closes modal; renders Modal/Input with correct key', () => {
```
Comment (julianguyen) [inline client/app/widgets/QuickCreate/_tests_/QuickCreate.spec.jsx]: Let's rephrase this to 'opens the accordion, closes the modal, and assigns the correct key to the modal and input field'.

**Comment by julianguyen** [comment]
Looks like you'll have to rebase before merging!

**Comment by thumbsupep** [comment]
seems like I do need help. I synced and started to rebase `issue_1158` from my master branch, but am unsure how to move forward resolving conflicts when its a file I didnt change.

**Comment by julianguyen** [comment]
@thumbsupep What does your console look like when you do `git rebase master` on your branch?

**Comment by julianguyen** [comment]
Doing a squash commit would actually bypass the the rebase issue!

**Comment by thumbsupep** [comment]
thanks!
